pkg/field/context.go

```diff
@@ -10,8 +10,10 @@ const ctxKey = contextKey(43)
 
 // FromContext returns the fields present in ctx if any
 func FromContext(ctx context.Context) Fields {
-	if s, ok := ctx.Value(ctxKey).(Fields); ok {
-		return s
+	if ctx != nil {
```

Comment:
OTOH, this should be handled by the caller. For example, `context.WithDeadline()` does not accept a `nil` context.
However, this was not explicitly documented in `FromContext` from the beginning, and overall this approach is more robust.